### DISCUSSION TITLE
Nativewind and Emotion interoperability

### DISCUSSION THREAD

**Discussion by yuriy-yarosh** [nativewind/nativewind#669] [Ideas]
[Emotion](https://emotion.sh) uses the same approach with importSource [babel preset](https://emotion.sh/docs/css-prop#babel-preset), and is very handy for critical css [webapp extraction](https://emotion.sh/docs/ssr#on-server).

Nativewind could potentially wrap the respective emotion jsxImports for interoperability.

[Twin.macro](https://github.com/ben-rogerson/twin.macro) is a very handy tailwind wrapper, uses the same babel macro approach and should work out of the box.

**Comment by marklawlor** [comment]
I'm the only contributor and I'm flat out just supporting Tailwind CSS. Support Emotion and Twin are possible (probably as a new project), but I highly doubt this will happen without support from the community.

The underlying tech being NativeWind isn't tied to Tailwind CSS, so any compiled CSS solution can work. I have an example here that uses vanilla extract https://github.com/marklawlor/neapolitan-extract

If someone produce a prototype using Emotion/Twin, I'd be happy to advise on the project.
